File: 2k17/despojados.py
import sys, math
n = input ()

'''
def prime (a):
  for i in range (2, int(math.sqrt (a))):
    if (a % i == 0):
      return False
  return True

ans = 0
count = 0
i = 2
while i < math.sqrt (n):
  if (prime(i)):
    if (count == 1):
      ans += 1
    if (aux % i == 0):
      aux /= i
      count += 1
      continue
  count = 0
  i += 1
'''
ms = {1}
def factors (n):
  x = 2
  while (x * x) <= n:
    while (n % x == 0):
      ms.add (x)
      n /= x
    x += 1
  if n > 1:
    ms.add (n)

factors (n)
ans = len (ms) - 1

ans = int(math.pow (2, ans) - (ans + 1))

print (str(ans))

File: 2k17/despojados.cpp
#include <bits/stdc++.h>

int prime (long long a) {
  for (int i = 2; i < a; i++)
    if (a % i == 0)
      return 0;
  return 1;
}

int main () {
  long long unsigned n, aux;

  scanf ("%llu", &n);
  aux = n;

  long long ans = 0, count = 0;
  for (long long i = 2; i < sqrt (n); i++) {
    if (prime (i)) {
      if (count == 1)
        ans++;
      if (aux % i == 0) {
        aux /= i;
        i--;
        count++;
        continue;
      }
    }
    count = 0;
  }

  printf ("%lld\n", ans);

  return 0;
}
